src/node/stage/approved.rs

```diff
@@ -676,10 +678,17 @@ impl Approved {
             .prefix
             .is_neighbour(self.shared_state.our_prefix())
         {
-            self.cast_unordered_vote(core, Vote::SectionInfo(elders_info))?;
+            self.cast_unordered_vote(core, Vote::SectionInfo(elders_info.clone()))?;
         }
 
-        Ok(())
+        if let Some(dst_key) = dst_key {
+            if dst_key == *self.shared_state.our_history.last_key() {
+                // The sender's knowledge of us is latest.
+                return Ok(());
+            }
+        }
+
+        self.send_neighbour_info(core, elders_info.prefix, nonce, Some(src_key))
```

Comment:
Wonder if this whole part is necessary. Isn't this going to be handled by `update_section_knowledge` triggered by the reception of this `NeighbourInfo` message?

Reply:
Ah, it was required maybe because I improperly populated the dst_key field previously.

Got this block of code removed.